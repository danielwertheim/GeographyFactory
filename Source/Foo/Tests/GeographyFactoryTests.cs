using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace Foo.Tests
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class GeographyFactoryTests
    {
        [Test]
        public virtual void When_Coordinates_has_correct_order_and_is_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.CorrectOrderAndClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_correct_order_and_is_not_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.CorrectOrderNotClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_wrong_order_and_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.WrongOrderAndClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_wrong_order_and_not_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.WrongOrderNotClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Coordinates_has_wrong_order_starting_south_to_north_and_many_to_west_and_back_and_is_closed()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.WrongSouthToNorthAndManyWestAndBackClosed());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Castle()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.Castle());

            Approvals.Verify(g);
        }

        [Test]
        public virtual void When_Star_drawn_from_middle_and_clockwise()
        {
            var g = GeographyFactory.CreatePolygon(CoordinatesTestFactory.StartThatDoesNotRenderWithoutFix());

            Approvals.Verify(g);
        }
    }
}