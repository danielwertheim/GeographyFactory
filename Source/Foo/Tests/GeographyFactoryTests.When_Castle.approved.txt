POLYGON ((9.5204846858978343 49.758400165268135, 9.520565152168281 49.758663540943289, 9.520715355873115 49.758826417289669, 9.520715355873115 49.758954638709376, 9.520940661430366 49.759068998067924, 9.520865559577949 49.759172960887113, 9.5206885337829661 49.759211080531635, 9.5206187963485789 49.759328904697945, 9.5203827619552683 49.759315043046179, 9.5201735496521067 49.759249200146179, 9.5198945999145579 49.759235338471633, 9.51957273483277 49.759335835522336, 9.51936352252961 49.75936702421987, 9.51910603046418 49.759134841212628, 9.5190309286117625 49.758933846069759, 9.5191274881362986 49.758760573726164, 9.5192884206771922 49.758604628087532, 9.51936352252961 49.758424423613825, 9.5194815397262644 49.75831006273507, 9.5197819471359324 49.758251149449848, 9.5204846858978343 49.758400165268135))